doc/api/events.md

```diff
@@ -935,13 +935,415 @@ if the `EventEmitter` emits `'error'`. It removes all listeners when
+   to closely emulate a subset of the `EventEmitter` API.
+
+### `EventTarget` vs. `EventEmitter`
+
+The Node.js `EventTarget` object implements a modified subset of the
+`EventEmitter` API that allows it to closely emulate an `EventEmitter` in
+certain situations. It is important to understand, however, that an
+`EventTarget` is *not* an instance of `EventEmitter` and cannot be used in
+place of an `EventEmitter` in most cases.
+
+1. Unlike `EventEmitter`, any given `listener` can be registered at most once
+   per event `type`. Attempts to register a `listener` multiple times will be
+   ignored.
+2. While the Node.js `EventTarget` exposes an `emit()` function that uses the
+   same signature as `EventEmitter`, the event will be dispatched to registered
+   listeners as a single event object composed as an array with an additional
+   `type` property rather than as separate arguments.
+3. The Node.js `EventTarget` does not emulate the full `EventEmitter` API.
+   Specifically the `prependListener()`, `prependOnceListener()`,
+   `rawListeners()`, `setMaxListeners()`, `getMaxListeners()`, and
+   `errorMonitor` APIs are not emulated.
+4. The Node.js `EventTarget` does not implement any special default behavior
+   for events with type `'error'`.
+5. The Node.js `EventTarget` supports `EventListener` objects as well as
+   functions as handlers for all event types.
+
+### Event Listener
+
+Event listeners registered for an event `type` may either be JavaScript
+functions or objects with a `handleEvent` property whose value is a function.
+
+In either case, the handler function will be invoked with the `event` argument
+passed to the `eventTarget.dispatchEvent()` function.
+
+Async functions may be used as event listeners. If an async handler function
+rejects, the rejection will be captured and be will handled as described in
+[`EventTarget` Error Handling][].
+
+An error thrown by one handler function will not prevent the other handlers
+from being invoked.
+
+The return value of a handler function will be ignored.
+
+Handlers are always invoked in the order they were added.
+
+Handler functions may mutate the `event` object.
+
+```js
+function handler1(event) {
+  console.log(event.type);  // Prints 'foo'
+  event.a = 1;
+}
+
+async function handler2(event) {
+  console.log(event.type);  // Prints 'foo'
+  console.log(event.a);  // Prints 1
+}
+
+const handler3 = {
+  handleEvent(event) {
+    console.log(event.type);  // Prints 'foo'
+  }
+};
+
+const handler4 = {
+  async handleEvent(event) {
+    console.log(event.type);  // Prints 'foo'
+  }
+};
+
+const target = new EventTarget();
+target.addEventListener('foo', handler1);
+target.addEventListener('foo', handler2);
+target.addEventListener('foo', handler3);
+target.addEventListener('foo', handler4, { once: true });
+
+target.dispatchEvent({ type: 'foo' });
+```
+
+When the `eventTarget.emit()` function is used. the arguments passed
```

Comment:
Why would we want to ship these sort of extensions? Is this so it's easy to write "polymorphic" code that pretends an EventTarget is an EventEmitter?

I prefer to rely on the static methods here - IIRC `events.on` already supports EventTarget and `.once` _should_ (I don't remember if it does)

Reply:
Yes, this is purely to allow an isomorphic approach with `EventEmitter`. I consider the `emit()` here to be optional and I'm fine with omitting it if folks don't think it's necessary.

Reply:
Given that we're not exposing the `EventTarget` constructor, this makes less sense to expose the `emit()`... removed!